src/pyscript/plugins/create.py

```diff
@@ -20,25 +27,111 @@ def create(
+    directory in the current directory. Alternatively, use `--wrap` so as to embed
+    a python file instead.
     """
-    try:
-        create_project(
-            app_name,
-            app_description,
-            author_name,
-            author_email,
-            pyscript_version,
-            project_type,
+    if not app_or_file_name and not command:
+        raise cli.Abort(
+            "Must provide either an input '.py' file or a command with the '-c' option."
         )
-    except FileExistsError:
+
+    if app_or_file_name and command:
+        raise cli.Abort("Cannot provide both an input '.py' file and '-c' option.")
+
+    if (title or output or show or command) and (not wrap):
         raise cli.Abort(
-            f"A directory called {app_name} already exists in this location."
+            """`--title`, `--output/-o`, `--show` and `--command/-c`
+            are meant to be used with `--wrap/-w`"""
         )
 
+    if not wrap:
+        try:
+            if not app_description:
+                app_description = typer.prompt("App description")
+            if not author_name:
+                author_name = typer.prompt("Author name")
+            if not author_email:
+                author_email = typer.prompt("Author email")
+            create_project(
+                str(app_or_file_name),
+                app_description,
+                author_name,
+                author_email,
+                pyscript_version,
+                project_type,
+            )
+        except FileExistsError:
+            raise cli.Abort(
+                f"A directory called {app_or_file_name} already exists in this location."
+            )
+    else:
+        title = title or "PyScript App"
+
+        # Derive the output path if it is not provided
+        remove_output = False
+        if output is None:
+            if command and show:
+                output = Path("pyscript_tmp.html")
+                remove_output = True
+            elif not command:
+                assert app_or_file_name is not None
+                output = app_or_file_name.with_suffix(".html")
+            else:
+                raise cli.Abort("Must provide an output file or use `--show` option")
+        if app_or_file_name is not None:
+            file_to_html(
+                app_or_file_name,
+                title,
+                output,
+                template_name="wrap.html",
+                pyscript_version=pyscript_version,
+            )
+        if command:
+            string_to_html(
+                command,
+                title,
+                output,
+                template_name="wrap.html",
+                pyscript_version=pyscript_version,
+            )
+        if output:
+            if show:
+                console.print("Opening in web browser!")
+                webbrowser.open(f"file://{output.resolve()}")
+            if remove_output:
+                time.sleep(1)
+                output.unlink()
```

Comment:
Just a thought, perhaps this logic could be placed inside a function? This if/else is a bit hard to follow

Reply:
Yeah, thanks!